Http/Controllers/API/V1/PriceController.php

```diff
@@ -23,75 +24,77 @@ public function index(Request $request)
         }
 
-        if (count($lists)) {
-            foreach ($lists as $key => $list) {
-                $list = is_numeric($list) ? TenantPriceList::find($list) : $list;
-                $this->authorize('view', [$list, TenantPriceList::class]);
-                $lists[$key] = $list->id;
-            }
-        } else {
-            $this->authorize('view', [null, TenantPriceList::class]);
+        if ($request->has('sort_by')) {
+            $query->orderBy('prices.'.$request->get('sort_by'), $request->get('order') ?: 'asc');
         }
 
-        $query = Price::only()
-            ->forPriceLists($lists)
-            ->when($request->has('deleted'), function ($query) use ($request) {
-                return $query->where('prices.deleted', $request->boolean('deleted'));
-            });
+		return $this->paginator($query->paginate($request->get('per_page') ?: 30));
+    }
 
-        if ($request->has('code')) {
-            $query->withSocialInsuranceAgencyId($request->get('code'));
+    /**
+     * Builds an array of List IDs from request parameters.
+     *
+     * @param  \Illuminate\\Http\Request  $request
+     * @return array
+     */
+    private function buildListsArrayFromRequestParams(Request $request){
 
-            if ($price = $query->first()) {
-                return $this->item($price);
-            }
+        $lists = [];
 
-            return $this->emptyItem();
+        if ($request->has('list_id')) {
+            $lists[] = $request->get('list_id');
+            return $lists;
         }
 
-        if ($request->has('price_code')) {
-            $query->where('price_code', $request->get('price_code'));
-            if ($price = $query->first()) {
-                return $this->item($price);
+        if ($request->has('list_ids')) {
+            $lists = [];
+            foreach ( $request->array('list_ids') as $key => $listId ){
+                $lists[] = $listId;
             }
-
-            return $this->emptyItem();
+            return $lists;
         }
 
-        if ($request->has('query')) {
-            $query->leftJoin(TenantDbHelper::getMasterDbAlias().'.price_list_references', 'price_list_references.id', '=', 'prices.state_price_id');
-
-            $query = $this->addQueryClause($request, $query, [
-                'prices.description',
-                'prices.price_code',
-                'price_list_references.social_insurance_agency_id',
-                'price_list_references.description',
-            ]);
-
-            $this->addRelevanceOrderClause($request, $query, [
-                'prices.price_code',
-                'prices.description',
-            ]);
+        if ($request->has('reimbursement_type_id')) {
+            $query = TenantPriceList::forReimbursementType($request->get('reimbursement_type_id'))
+                ->used();
+            if ($priceList = $query->first()) {
+                $lists[] = $priceList->id;
```

Comment:
again we keep the format consisted by just adding numerical IDs to the $lists array. This means no later conversions or parsing of $lists is needed.